client/src/lib/RenderReviewList.js

```diff
@@ -0,0 +1,219 @@
+import DataTable, {createTheme} from 'react-data-table-component';
+import React from 'react';
+import {format} from 'date-fns';
+import Select from 'react-select';
+
+const RenderReviewList = () => {
+    let lastFetch;
+    const queryString = new URLSearchParams(window.location.search);
+    const status = queryString.get('status') ?? 'pending';
+    const [data, setData] = React.useState([]);
+    const [loading, setLoading] = React.useState(false);
+    const [totalRows, setTotalRows] = React.useState(0);
+    const [perPage, setPerPage] = React.useState(10);
+    const [currentPage, setCurrentPage] = React.useState(1);
+    const [channelOptions, setChannelOptions] = React.useState([]);
+    const [selectedChannel, setSelectedChannel] = React.useState(queryString.get('channel') ?? 'all');
+    const [selectPlaceHolder, setSelectPlaceHolder] = React.useState('Code Reviews For All Slack Channels');
+
+    const fetchData = async ({channel, limit, page, status, description}) => {
+        // Check last fetch to avoid data refresh during first page loading.
+        if (lastFetch && lastFetch === `${channel}-${limit}-${page}-${status}`) {
+            return;
+        }
+        lastFetch = `${channel}-${limit}-${page}-${status}`;
+        console.log('fetchData', description, lastFetch)
+        setLoading(true);
+        fetch(`/api/reviews/${channel}/${status}?limit=${limit}&page=${page}`)
+            .then((res) => res.json())
+            .then((result) => {
+                setData(result.dataset);
+                setTotalRows(result.total);
+                setLoading(false);
+            });
+    };
+
+    const handlePageChange = page => {
+        console.log('handlePageChange', page);
+        setCurrentPage(page);
+        fetchData({
```

Comment:
fetchData( { ...
can be defined as separate method and called from 39, 51, 66 , 104 lines

Reply:
The fetchData call signature was implement this way to allow finer control on how each data is fetch base on their respective data change after and before the side affects trigger.  Prior version had a combine reused side affects variables, prove to be harder to troubleshoot and improve when addition features are added because each state cause by side affects will cause data to refresh and depending on how reach and ordered of evens fired, the incorrect data is called by fetchData.

Reply:
Thank, you for your wide response.